pkg/controller/backup_finalizer_controller.go

```diff
@@ -209,3 +250,167 @@ func (r *backupFinalizerReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		For(&velerov1api.Backup{}).
 		Complete(r)
 }
+
+func (r *backupFinalizerReconciler) listVolumeSnapshotsToDelete(backupName string) ([]snapshotv1api.VolumeSnapshot, error) {
+	volumeSnapshots := make([]snapshotv1api.VolumeSnapshot, 0)
+
+	if r.volumeSnapshotLister != nil {
+		tmpVSs, err := r.volumeSnapshotLister.List(label.NewSelectorForBackup(backupName))
+		if err != nil {
+			r.log.Error(err)
+			return volumeSnapshots, err
+		}
+		for _, vs := range tmpVSs {
+			volumeSnapshots = append(volumeSnapshots, *vs)
+		}
+	}
+	return volumeSnapshots, nil
+}
+
+// deleteVolumeSnapshots delete VolumeSnapshot created during backup.
+// This is used to avoid deleting namespace in cluster triggers the VolumeSnapshot deletion,
+// which will cause snapshot deletion on cloud provider, then backup cannot restore the PV.
+// If DeletionPolicy is Retain, just delete it. If DeletionPolicy is Delete, need to
+// change DeletionPolicy to Retain before deleting VS, then change DeletionPolicy back to Delete.
+func (r *backupFinalizerReconciler) deleteVolumeSnapshots(volumeSnapshots []snapshotv1api.VolumeSnapshot,
```

Comment:
The functions that are no longer called from backup_controller should probably be deleted there as well.

Reply:
They are in use at other places